cpp/src/arrow/csv/writer_test.cc

```diff
@@ -240,6 +241,10 @@ std::vector<WriterTestParams> GenerateTestCases() {
       reject_structural_params({nullptr, "a", nullptr, "c,d", nullptr, ",e", "f"}, "c,d"),
       reject_structural_params({"a", "b", nullptr, "c,d", nullptr, nullptr}, "c,d"),
       // exercise simd code with strings total length >= 16
+      {abc_schema, populated_batch,
+       DefaultTestOptions(/*include_header=*/false, "", QuotingStyle::AllValid, "\n", ',',
+                          /*batch_size=*/100),
```

Comment:
Process test data [1] as a single batch (batch_size > rows) so the string array size > 16.
[1] https://github.com/apache/arrow/blob/master/cpp/src/arrow/csv/writer_test.cc#L101